pcicrawler/ocp_diag.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python3
+
+"""
+Copyright (c) Meta Platforms, Inc. and affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+import json
+import socket
+import pkg_resources
+import typing as ty
+import ocptv.output as tv
+from pci_lib import PCIDevice
+from abc import abstractmethod
+
+
+class OCPTestStep:
+    def __init__(self,
+                    device: PCIDevice,
+                    expected_result: str,
+                    ocp_run: tv.TestRun):
+        self._device = device
+        self._expected_result = expected_result
+        self._ocp_run = ocp_run
+        self._error_messages = []
+
+    @property
+    @abstractmethod
+    def _test_name(self):
+        pass
+
+    @abstractmethod
+    def run(self, current_value: str):
+        step = self._ocp_run.add_step(self._test_name)
+        if str(current_value) != str(self._expected_result):
+            step.add_diagnosis(tv.DiagnosisType.FAIL, 
+                            verdict="Test {} for device {}, expected {}, found {}.".format(
+                                self._test_name,
+                                self._device,
+                                self._expected_result,
+                                str(current_value)))
+            return False
+        else:
```

Comment:
`else:` is not needed; you have a `return False` in the first branch

Reply:
coding nitpick lol

Reply:
anything to make the code easier to read; code is for humans